src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -179,32 +176,42 @@ private void flush() {
         } finally {
             rwlock.writeLock().unlock();
         }
-        final Iterator<Pair<Record<GenericRecord>, Blob>> iter = recordsToInsert.iterator();
-
-        while (iter.hasNext()) {
-            final Pair<Record<GenericRecord>, Blob> blobPair = iter.next();
-            try {
-                log.info("upload blob {}", blobPair.getLeft().getRecordSequence().get());
-                blobStore.putBlob(sinkConfig.getBucket(), blobPair.getRight(), PutOptions.NONE);
-                blobPair.getLeft().ack();
-                log.info("write success {}", blobPair.getLeft().getRecordSequence().get());
-            } catch (ContainerNotFoundException e) {
-                log.error("Bad message", e);
-                blobPair.getLeft().fail();
-                iter.remove();
-            }
+        Record<GenericRecord> firstRecord = recordsToInsert.get(0);
+        Schema<GenericRecord> schema = getPulsarSchema(firstRecord);
+        format.initSchema(schema);
+
+        final Iterator<Record<GenericRecord>> iter = recordsToInsert.iterator();
+        try {
+            String filepath = buildPartitionPath(firstRecord, partitioner, format);
```

Comment:
why use `firstRecord` to get the `filepath`? what if batch messages with different partition path? like if we have a time based partitioner with hours, and a batch message contains messages from 00:59 ~ 01:01, shouldn't the messages be separated into two partition file?

Reply:
Since the file is written once, the data needs to be divided into sections. For example, when writing by time, the time is used as the path, and the file name is the unique id of the first message of each batch of messages.
example：
`2021/05/18/17/0099887.json`